lib/llm/src/local_model.rs

```diff
@@ -391,10 +402,40 @@ impl LocalModel {
         // Publish the Model Deployment Card to etcd
         let kvstore: Box<dyn KeyValueStore> = Box::new(EtcdStorage::new(etcd_client.clone()));
         let card_store = Arc::new(KeyValueStoreManager::new(kvstore));
-        let key = self.card.slug().to_string();
-        card_store
-            .publish(model_card::ROOT_PATH, None, &key, &mut self.card)
-            .await?;
+        let slug_key = self.card.slug();
+        let key = slug_key.to_string();
+
+        // Try to load existing MDC to determine if update is needed
+        match card_store.load::<ModelDeploymentCard>(model_card::ROOT_PATH, slug_key).await {
```

Comment:
I think we do want to do something like this, but not here. Could you extract it into a different PR?